Cython/Compiler/PyrexTypes.py

```diff
@@ -2683,6 +2683,13 @@ def __init__(self, base_type):
                 self.from_py_function = "__Pyx_PyUnicode_AsUnicode"
             self.exception_value = "NULL"
 
+    def create_to_py_utility_code(self, env):
+        if self.to_py_function == "__Pyx_PyUnicode_FromUnicode":
+            # Not included in the standard TypeConversions.c string conversions because Py_UNICODE
+            # is deprecated.
+            env.use_utility_code(UtilityCode.load_cached("pyunicode_from_unicode", "StringTools.c"))
+        return super().create_to_py_utility_code(env)
+
```

Comment:
In order to avoid this special case (where, really, all of the conversion functions are special in their own way), I suggest adding a `to_py_utility_code` attribute in this class that can hold required utility code (`None` in the class's default), set at need by `__init__`, where we distinguish the cases already. `CTypeDefType` does something similar (but loads the code later when we know more about the context).

I also don't think we need to call `super()` if there's utility code available. Subclasses don't do that either.